content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to your models, result objects, queries, or to Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To use a Prisma Client extension, you must create an associated _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. The standard client does not change.
+
+You can use [multiple independent extensions at the same time](#multiple-extensions). You can run an extension in its own isolated extended client, or combine it with other extensions in an extended client.
+
+You can also [share your Prisma Client extensions](/concepts/components/prisma-client/client-extensions/shared-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+### Client instances
+
+Extended clients interact with each other, and with the standard client, as follows:
+
+- Each extended client operates independently in an isolated instance.
+- Extended clients cannot conflict with each other, or with the standard client.
```

Comment:
I think it would be worth adding here that when there's is a conflict, the last extension that has been hooked in always wins.

Reply:
would it even make sense to add a practical example here already (e.g. "For example, when two instances modify the `result` by adding a field with an identical name but different values. In this case, the last extension determines the value of that field.")

FWIW, I'm not convinced the best place for such an example would be right here but IMO we should certainly have such an example _somewhere_.

Reply:
@nikolasburk  We do cover that in this page.

It seems weird to me to say that they cannot conflict, but when they conflict _this_ happens... Maybe it would be better to say that conflicts are handled cleanly?

@millsp @nikolasburk  What do you think?

Also, what is it that conflicts here? The extensions themselves, or the extended clients?

Reply:
Ah, and from Pierre's comment above, it sounds like it might not be true now (in all cases?).

Reply:
Again, I think the terminology is confusing us here. This section is about extended clients - the forked client instances. It's not about extensions. I think once we've decided on a new term that doesn't involve the word "extended", the doc will be a lot clearer.